### PR TITLE
[Task-3] Setup CI

### LINKED ISSUE #3: [Task-3] Setup CI
### Acceptance Criteria
Setup CI to run the test on Github Actions

### PR DESCRIPTION
**Pull request by gutakk**
Resolve #3 

## What happened 👀

- Setup CI for both unit and UI test on Github Actions
- Ignore Cypress video result
 
## Insight 📝

- Run CI every on push
- Run UI test following the [Cypress documentation](https://docs.cypress.io/guides/continuous-integration/github-actions#Cypress-GitHub-Action)
 
## Proof Of Work 📹

CI workflow was passed https://github.com/gutakk/react-surveyor/actions/runs/905378780

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Test
+
+on: push
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
```
Comment (carryall) [inline .github/workflows/test.yml:9]: @gutakk How about added this to [cancel the previous ongoing test running](https://github.com/styfle/cancel-workflow-action) when we push a newer commit? 😄 
```
- name: Cancel previous runs
   uses: styfle/cancel-workflow-action@0.6.0
   with:
      access_token: ${{ github.token }}
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Stylelint
+        run: npm run stylelint
+
+      - name: Unit test
+        run: npm run test
+
+      - name: UI test
+        uses: cypress-io/github-action@v2
+        with:
+          start: npm start
```
Comment (rafayet-monon) [inline .github/workflows/test.yml:28]: How about adding a `wait-on: http://localhost:3000` here?

File: .github/workflows/test.yml (hunk 1 above)
Comment (gutakk) [reply .github/workflows/test.yml:9]: That was nice. Fixed in 0ee3b1889fb15e55515cd052742ff44ce133b8b5.

File: .github/workflows/test.yml (hunk 2 above)
Comment (gutakk) [reply .github/workflows/test.yml:33]: Nice catch. Fixed in 0ee3b1889fb15e55515cd052742ff44ce133b8b5.

**Review by rafayet-monon — APPROVED** [review]

**Review by carryall — APPROVED** [review]

**Pull request merged**